src/Cache/FileStore.php

```diff
@@ -204,17 +205,18 @@ public function set(string $key, $value)
         }
 
         return $this->lock($cacheFile, LOCK_EX, function ($handle) use ($cacheKey, $cacheFile, $value) {
-            $contents = fread($handle, filesize($cacheFile) ?: 1) ?? '';
+            $size = fstat($handle)['size'];
+            $contents = fread($handle, $size ?: 1) ?? '';
             $contents = json_decode($contents, true) ?? [];
 
             if ( ! empty($contents[$cacheKey]) && \is_array($value)) {
                 $contents[$cacheKey] = $value + $contents[$cacheKey];
             } else {
                 $contents[$cacheKey] = $value;
             }
-
-            return $this->put($cacheFile, json_encode($contents), self::LOCK_NONE);
-        });
+            ftruncate($handle, 0);
```

Comment:
Why do we need to truncate here?

Reply:
Since it's in concatenation mode, since write mode would format the file before any reads, it reads the file and needs to format to insert its new JSON

ftruncate in this case clean the file with $handle